src/util/domains.py

```diff
@@ -45,6 +43,95 @@
 except NameError:
     basestring = unicode = str
 
+# import from domains/expression
+try:
+    import odoo.domains as _dom
+
+    FALSE_LEAF = _dom._FALSE_LEAF
+    TRUE_LEAF = _dom._TRUE_LEAF
+    NOT_OPERATOR = _dom.DomainNot.OPERATOR
+    AND_OPERATOR = _dom.DomainAnd.OPERATOR
+    OR_OPERATOR = _dom.DomainOr.OPERATOR
+    DOMAIN_OPERATORS = {NOT_OPERATOR, AND_OPERATOR, OR_OPERATOR}
+
+    # normalization functions are redefined here because they will be deprecated
+    # the Domain factory normalized but also distributes operators during creation of domains
+
+    def normalize_domain(domain):
+        """Return a normalized version of the domain.
```

Comment:
Why don't we implement this in terms of the new `Domain` factory? 
`return list(_dom.Domain(domain))` would this be enough?
If not, I still believe it's better to use the _new_ implementation, it will bring better compatibility for future versions.

Reply:
it is not strictly equivalent and some tests were failing.
Here is an example from an older build: https://runbot.odoo.com/runbot/build/64886163
One of the differences is stated above: `list(Domain(xxx))` is `distribute_not(normalize_domain(xxx))`.

I suggest that we merge as-is, and I can implement a follow-up PR with the simplification.

Reply:
That failure doesn't make sense 🤯  (irrelevantly of the current implementation) it's a test that should only run when going over 17.3. Why is it trying to upgrade from 17 to master, that's not valid. Only >=18 can be upgraded to master. Checking with runbot team...

Reply:
@aj-fuentes Added to my TODO list

Reply:
Note that the last runbot run upgraded 18->master. Thus the tests that failed before, shouldn't anymore.